apps/openmw/mwgui/recharge.cpp

```diff
@@ -179,7 +177,9 @@ void Recharge::onItemClicked(MyGUI::Widget *sender, const MWWorld::Ptr& item)
     if (gem.getRefData().getCount() == 0)
     {
         std::string message = MWBase::Environment::get().getWorld()->getStore().get<ESM::GameSetting>().find("sNotifyMessage51")->mValue.getString();
-        message = boost::str(boost::format(message) % gem.getClass().getName(gem));
+        if (message.find("%s") != std::string::npos)
+            message.replace(message.find("%s"), 2, gem.getClass().getName(gem));
```

Comment:
Why don't put this logic into separate function? Each `boost::format` call now is replaced by `if`. This increases algorithmic complexity of each function. And because there is no function you have to call find two times to avoid new variable definition. If eventually c++ will get `std::format` for string it would be much easier to just change our function implementation than to find all of this if-find-replace-find patterns.

Reply:
Hm, okay. I'll put something into StringUtils.